### DISCUSSION TITLE
What are some advantages and disadvantages of fanout

### DISCUSSION THREAD

**Discussion by vordimous** [aklivity/zilla#1158] [Q&A] [answered]
@cadzchua asked
"can I ask about fanout and what are some advantages and disadvantages it brings. Should I implement it in my case?"

**Comment by vordimous** [comment, marked as answer]
Fanout in Zilla manages the number of clients that connect to a Kafka cluster and provides a caching layer. 

Thousands of unique clients may need to read or write data. Zilla maintains an active connection pool to Kafka with just a few connections open to support all the traffic from the different clients. 

The caching layer in Zilla will improve read times for all clients. Zilla natively stays updated with changes on the underlying Kafka topics, ensuring that all clients experience enhanced performance.

Another practical benefit of Fanout is that it removes the need for developers to understand the complexities of a Kafka topic. Zilla allows devs to stream data inside an application using standard protocols.

**Reply by cadzchua** [reply]
any disadvantage? if they are all advantage I guess I should implement it in my case?

**Reply by vordimous** [reply]
The only disadvantage is the added complexity to a tech stack. If you have only one small service interacting with Kafka and a small team that all generally knows how to produce and consume Kafka messages then Zilla could be overkill. Zilla will still do the job, but it might be less effort to just talk to Kafka directly. 

With that said, there is always the possibility that a project will grow to need Zilla, so having it in the beginning just means you don't have to go back and refactor code later.
